R/dp_deploy.R

```diff
@@ -22,30 +22,51 @@ dp_deploy <- function(project_path = ".", ...) {
   # get daap content and info
   conf <- dpconf_get(project_path = project_path)
   dlog <- get_dlog(project_path = project_path)
-  d <- readRDS(file = glue::glue(
-    "{project_path}/output_files/RDS_format/",
-    "data_object.RDS"
-  ))
+  # get the data object file type
+  type <- detect_type(project_path)
+  d <- object_read(project_path, type)
 
   dp_deployCore(
     conf = conf, project_path = project_path, d = d, dlog = dlog,
-    git_info = git_info, ...
+    git_info = git_info, type = type, ...
   )
 }
 
+detect_type = function(project_path){
```

Comment:
As mentioned in dpbuild PR: what if daap was started as rds and switched to qs? In this case the detected type will be wrong

Reply:
Please see my reply: [here](https://github.com/amashadihossein/dpbuild/pull/99#discussion_r1887745128)